### DISCUSSION TITLE
[5.0] (Re)consider adding a "Show full filenames in asset thumbnails" accessibility setting

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#14462] [Ideas] [answered]
I've suggested this [before](https://github.com/craftcms/cms/discussions/12236), but as Craft 5.0 is poised to introduce a brand new design for large thumbnails in asset fields, I thought the idea was worth bringing up again :)

In my opinion, the decision to (re)introduce non-truncated filenames in Craft 5's asset thumbnails, is a bit problematic.

Due to the fully visible filenames, the asset thumbnail elements in Craft 5 asset fields tend to vary a lot in width. This makes for a much messier layout than in Craft 4, that is harder to scan,  clunkier to sort, and less compact – with each thumbnail using more space than before (without the actual thumbnails themselves being noticeably bigger).

Here's an assets field with large thumbnails on Craft 4.7.4:

![CleanShot 2024-02-23 at 20 19 37@2x](https://github.com/craftcms/cms/assets/298510/dd7986a1-548a-4213-9902-7fd4e965d41e)

...and here's an equivalent field on Craft 5.0.0-beta.3:

![CleanShot 2024-02-23 at 20 22 12@2x](https://github.com/craftcms/cms/assets/298510/6117048b-ef94-46d8-899b-93d37cd17640)

The less compact design and uneven distribution can be especially noticeable in slideouts: 

![CleanShot 2024-02-23 at 20 23 44@2x](https://github.com/craftcms/cms/assets/298510/aea5f463-7aa7-4a81-9991-3b6ca82ae647)

![CleanShot 2024-02-23 at 20 23 16@2x](https://github.com/craftcms/cms/assets/298510/6aff73b9-6f7f-4d3a-a9d5-1e0fe334e4c1)

To me, "Large Thumbnails" feels like it has a relatively specific use case – i.e. to provide a view optimised for visual navigation, in cases where the user (or developer) feels that this will be the most effective way to manage the files. A lot of the time, filenames are typically trash – and in this context the tradeoff in visual coherence doesn't feel worth it to me, in most cases.

_Curiously, asset element indexes in the 5.0.0-beta.3 still use truncated filenames for the thumb view._ The element index seems like it's using a completely different UI element though, so maybe the new one just haven't been implemented there yet?

_I do appreciate that the full filenames has likely been introduced due to accessibility concerns._ Personally I think a new accessibility setting opting in to the non-truncated version would be a good solution; ideally this setting would also affect thumbs in asset element indexes, for a consistent user experience.

**Comment by thupsi** [comment]
+1 from me for making this optional. This impacts usability in a negative way for many users I think. A scannable and compact grid is good for everyone.

**Comment by brandonkelly** [comment, marked as answer]
We just released Craft 5.0.0-beta.6, which brings back asset title truncation, and shows tooltips on hover/focus with the full asset titles instead. (#14502)
